### PR TITLE
Sql db in Rust (without string passing)

### PR DESCRIPTION
**Pull request by RicoGit**
Here is a new example in Rust for WasmVm. It's simple Sql DB, it can works with only primitive types (int, double). Run like this: 

         sbt vm-sqldb/run

If all are fine, it will return:
```[info] [success] Total time: 3 s, completed 26.09.2018 14:19:28
[info] <run_example.sh> Run example sqldb.jar
[info] Starts for input file /Users/rico/dev/fluence/fluence/vm/examples/sqldb/target/wasm32-unknown-unknown/release/sqldb.wasm
[info] > select * from Quotations; 
[info] |   0.0|   0.0|6500.0|  10.0|
[info] |   1.0|   1.0| 450.0|  10.0|
[info] |   2.0|   2.0| 100.0|  10.0|
[info] |   3.0|   0.0|7000.0|  11.0|
[info] |   4.0|   1.0| 500.0|  11.0|
[info] |   5.0|   2.0|  80.0|  11.0|
[info] |   6.0|   1.0|6700.0|  12.0|
[info] |   7.0|   2.0| 400.0|  12.0|
[info] |   8.0|   3.0| 120.0|  12.0|
[info] > select * from Quotations where price = 6700; 
[info] |   6.0|   1.0|6700.0|  12.0|
[info] > select * from Quotations where price < 500; 
[info] |   1.0|   1.0| 450.0|  10.0|
[info] |   2.0|   2.0| 100.0|  10.0|
[info] |   5.0|   2.0|  80.0|  11.0|
[info] |   7.0|   2.0| 400.0|  12.0|
[info] |   8.0|   3.0| 120.0|  12.0|
[info] > select count(*) from Quotations; 
[info] |   9.0|
[info] > select count(*) from Quotations where symbol = 2; 
[info] |   3.0|
[info] > select avg(price) from Quotations; 
[info] |2427.777777777778|
[info] > select avg(price) from Quotations where symbol = 2; 
[info] |193.33333333333334|
[info] [SUCCESS] Execution Results.
[info] initState=ByteVector(32 bytes, 0x216056cb9b0a17b4f5461ccd0490b4d5ff064fb3f2b6e6b9daa4a4d2bb345a0f) 
[info] finishState=ByteVector(32 bytes, 0xd968f578ef2a427f15c0c3e515d1f929c92c2abfe61cea5e1721c791da5019d8)
[success] Total time: 79 s, completed 26.09.2018 14:19:31
```

### CODE REVIEW

File: vm/examples/counter/src/main/rust/main.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+#[no_mangle]
+pub unsafe fn get() -> i64 {
+    COUNTER_.get()
+}
+
+fn main() {
+    // do nothing
+}
```
Comment (DieMyst) [inline vm/examples/counter/src/main/rust/main.rs]: EOF

File: vm/examples/counter/src/main/scala/fluence/vm/examples/CounterRunner.scala [hunk 2]
```diff
@@ -52,7 +52,7 @@ object CounterRunner extends IOApp {
 
     program.value.map {
       case Left(err) ⇒
-        println(s"[Error]: $err")
+        println(s"[Error]: $err cause=${err.getCause}")
```
[the remark this answers is not in the document]
Comment (RicoGit) [reply vm/examples/counter/src/main/scala/fluence/vm/examples/CounterRunner.scala]: It's a standalone CLI app, there is no need logger.

File: vm/examples/sqldb/src/main/rust/sql_db.rs [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        compare_mode: i8
+    ) -> Table<f64> {
+        Table::build(
+            self.table.rows.clone().into_iter()
+                .filter(|row| {
+                    match compare_mode {
+                        mode if mode < 0 => row[field_idx] < field_val,
+                        mode if mode == 0 => row[field_idx] == field_val,
+                        mode if mode > 0 => row[field_idx] > field_val,
+                        _ => panic!("impossible invariant"),
```
[the remark this answers is not in the document]
Comment (RicoGit) [reply vm/examples/sqldb/src/main/rust/sql_db.rs]: Yes, but I found a way to fix this.

File: vm/examples/sqldb/src/main/rust/main.rs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+#[no_mangle]
+pub fn next_row() -> i32 {
+    let mut db = sql_db::DB.lock().expect("Can't take a Database.");
+    db.next_row()
+}
+
+/// Returns a next field value for the current row
+#[no_mangle]
+pub fn next_field() -> f64 {
+    let mut db = sql_db::DB.lock().expect("Can't take a Database.");
```
[the remark this answers is not in the document]
Comment (RicoGit) [reply vm/examples/sqldb/src/main/rust/main.rs]: fixed.

**Pull request merged**